t/apicast-policy-tls_validation.t

```diff
@@ -0,0 +1,150 @@
+use lib 't';
+use Test::APIcast::Blackbox 'no_plan';
+
+env_to_apicast(
+    'APICAST_HTTPS_PORT' => "$Test::Nginx::Util::ServerPortForClient",
+    'APICAST_HTTPS_CERTIFICATE' => "$Test::Nginx::Util::ServRoot/html/server.crt",
+    'APICAST_HTTPS_CERTIFICATE_KEY' => "$Test::Nginx::Util::ServRoot/html/server.key",
+    'APICAST_HTTPS_SESSION_REUSE' => 'on',
+);
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: TLS Client Certificate is whitelisted and valid
+--- configuration eval
+use JSON qw(to_json);
+use File::Slurp qw(read_file);
+
+to_json({
+  services => [{
+    proxy => {
+        policy_chain => [
+          { name => 'apicast.policy.tls_validation',
+            configuration => {
+              whitelist => [
+                { pem_certificate => CORE::join('', read_file('t/fixtures/CA/client.crt')) }
+              ]
+            }
+          },
+          { name => 'apicast.policy.echo' },
+        ]
+    }
+  }]
+});
+--- test env
+proxy_ssl_certificate $TEST_NGINX_SERVER_ROOT/html/client.crt;
+proxy_ssl_certificate_key $TEST_NGINX_SERVER_ROOT/html/client.key;
+proxy_pass https://$server_addr:$apicast_port/t;
+proxy_set_header Host localhost;
+log_by_lua_block { collectgarbage() }
```

Comment:
Why is this needed?

Reply:
To run the Test::Nginx memory leak test mode. I was worried that there are some memory leaks in the OpenSSL wrapper, so I was running the memory leak checker.